### ISSUE TITLE
SurfR

### ISSUE THREAD

**Issue by auroramaurizio** [Bioconductor/Contributions#3105]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/auroramaurizio/SurfR

Confirm the following by editing each check box to '[x]'

- [X] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [X] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [X] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [X] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [X] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [X] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [X] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [X] The 'devel' branch for new packages and features.
- [X] The stable 'release' branch, made available every six
      months, for bug fixes.
- [X] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by auroramaurizio** [comment]
Dear @PeteHaitch,

Thank you for the feedback.

We fixed the linux-platform issue in the vignette and submitted the updated version.

We have tested the package on a local Mac OS computer and the R CMD check succeeded. At the end of this message we report the sessionInfo().
Unfortunately, we could not reproduce the mac-related error highlighted in the lconway BUILD SRC output: "Rhdf5lib was not compiled with support for the S3 VFD". We read the chat between you and a former package submitter addressing the same topic https://github.com/Bioconductor/Contributions/issues/2753 (and checked the package source) as well as several posts by Mike Smith such as https://support.bioconductor.org/p/9134972/#9153888. It seems more like a macOS-Rhdf5 related problem. What's your advice?

Best wishes,

Aurora and Anna Sofia


> sessionInfo()
R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6.3
 
Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
 
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
 
time zone: Europe/Rome
tzcode source: internal
 
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base    
 
other attached packages:
[1] stringr_1.5.0       SPsimSeq_1.10.0     SurfR_0.99.0        BiocManager_1.30.22
 
loaded via a namespace (and not attached):
  [1] splines_4.3.1               bitops_1.0-7                filelock_1.0.2              tibble_3.2.1                cellranger_1.1.0          
  [6] preprocessCore_1.62.1       XML_3.99-0.14               rpart_4.1.19                lifecycle_1.0.3             fastcluster_1.2.3         
 [11] edgeR_3.42.4                doParallel_1.0.17           lattice_0.21-8              MASS_7.3-60                 backports_1.4.1           
 [16] magrittr_2.0.3              openxlsx_4.2.5.2            limma_3.56.2                Hmisc_5.1-0                 rmarkdown_2.24            
 [21] zip_2.3.0                   DBI_1.1.3                   ade4_1.7-22                 abind_1.4-5                 zlibbioc_1.46.0           
 [26] rvest_1.0.3                 GenomicRanges_1.52.0        purrr_1.0.2                 metaRNASeq_1.0.7            BiocGenerics_0.46.0       
 [31] RCurl_1.98-1.12             phyloseq_1.44.0             nnet_7.3-19                 WriteXLS_6.4.0              rappdirs_0.3.3            
 [36] GenomeInfoDbData_1.2.10     IRanges_2.34.1              S4Vectors_0.38.1            ggrepel_0.9.3               vegan_2.6-4               
 [41] venn_1.11                   fitdistrplus_1.1-11         permute_0.9-7               codetools_0.2-19            DelayedArray_0.26.7       
 [46] xml2_1.3.5                  tidyselect_1.2.0            TCGAbiolinksGUI.data_1.20.0 matrixStats_1.0.0           stats4_4.3.1              
 [51] BiocFileCache_2.8.0         dynamicTreeCut_1.63-1       base64enc_0.1-3             jsonlite_1.8.7              multtest_2.56.0           
 [56] Formula_1.2-5               survival_3.5-7              iterators_1.0.14            foreach_1.5.2               tools_4.3.1               
 [61] progress_1.2.2              rio_0.5.30                  Rcpp_1.0.11                 glue_1.6.2                  gridExtra_2.3             
 [66] admisc_0.33                 xfun_0.40                   mgcv_1.9-0                  DESeq2_1.40.2               MatrixGenerics_1.12.3     
 [71] GenomeInfoDb_1.36.2         dplyr_1.1.3                 fastmap_1.1.1               rhdf5filters_1.12.1         fansi_1.0.4               
 [76] digest_0.6.33               R6_2.5.1                    colorspace_2.1-0            GO.db_3.17.0                biomaRt_2.56.1            
 [81] RSQLite_2.3.1               utf8_1.2.3                  tidyr_1.3.0                 generics_0.1.3              data.table_1.14.8         
 [86] prettyunits_1.1.1           httr_1.4.7                  htmlwidgets_1.6.2           S4Arrays_1.0.6              pkgconfig_2.0.3           
 [91] gtable_0.3.4                blob_1.2.4                  impute_1.74.1               SingleCellExperiment_1.22.0 XVector_0.40.0            
 [96] htmltools_0.5.6             biomformat_1.28.0           scales_1.2.1                Biobase_2.60.0              png_0.1-8                 
[101] enrichR_3.2                 knitr_1.43                  rstudioapi_0.15.0           tzdb_0.4.0                  reshape2_1.4.4            
[106] rjson_0.2.21                checkmate_2.2.0             nlme_3.1-163                curl_5.0.2                  cachem_1.0.8              
[111] rhdf5_2.44.0                parallel_4.3.1              foreign_0.8-84              AnnotationDbi_1.62.2        pillar_1.9.0              
[116] grid_4.3.1                  vctrs_0.6.3                 dbplyr_2.3.3                cluster_2.1.4               htmlTable_2.4.1           
[121] evaluate_0.21               TCGAbiolinks_2.28.3         readr_2.1.4                 mvtnorm_1.2-3               cli_3.6.1                 
[126] locfit_1.5-9.8              compiler_4.3.1              rlang_1.1.1                 crayon_1.5.2                assertr_3.0.0             
[131] plyr_1.8.8                  forcats_1.0.0               stringi_1.7.12              WGCNA_1.72-1                BiocParallel_1.34.2       
[136] munsell_0.5.0               Biostrings_2.68.1           Matrix_1.6-1                hms_1.1.3                   bit64_4.0.5               
[141] ggplot2_3.4.3               Rhdf5lib_1.22.0             KEGGREST_1.40.0             SummarizedExperiment_1.30.2 haven_2.5.3               
[146] igraph_1.5.1                memoise_2.0.1               bit_4.0.5                   readxl_1.4.3                downloader_0.4            
[151] ape_5.7-1

**Comment by auroramaurizio** [comment]
Dear @PeteHaitch,
It seems there might be a discrepancy between the package version in the build report and the actual version in the repositories.
Looking at the build report we noticed that the check is still made on package version 0.99.1 while the current version is 0.99.3. 
We pushed the latest version on both origin and upstream repos. 
Cloning https://git.bioconductor.org/packages/SurfR.git we see that everithing is up to date: the corrections have been implemented (removing SurfR.Proj file and updating .gitignore) and the version in DESCRIPTION is the latest 0.99.3.
Only the check on the build report is made on the old version 0.99.1.

Thank you for your attention to this matter. 

Best,
Aurora and Anna Sofia

**Comment by PeteHaitch** [comment]
@lshep any idea about why an old version is getting picked up by the build system (see https://github.com/Bioconductor/Contributions/issues/3105#issuecomment-1715127737)

**Comment by lshep** [comment]
It looks like something odd is going in; in particular also with the double build report kick off.  yet we don't see this with other repositories.  how exactly are you making changes and pushing to git.bioconductor.org?

**Comment by auroramaurizio** [comment]
Dear @lshep,
Thank you for your quick reply.

Here the commands:

git remote -v

#origin	git@github.com:auroramaurizio/SurfR.git (fetch)
#origin	git@github.com:auroramaurizio/SurfR.git (push)
#upstream	git@git.bioconductor.org:packages/SurfR.git (fetch)
#upstream	git@git.bioconductor.org:packages/SurfR.git (push)

git fetch --all
git merge origin/master
git merge upstream/master
git add .
git commit -m "bumped the package version 0.99.3"
git push origin master
git push upstream master
git push upstream master:devel

I think the double report kick off is due to the fact that I pushed to master instead of devel at first.
I got this message in the last commit:

git push upstream master

#Counting objects: 5, done.
#Delta compression using up to 8 threads.
#Compressing objects: 100% (3/3), done.
#Writing objects: 100% (3/3), 302 bytes | 0 bytes/s, done.
#Total 3 (delta 2), reused 0 (delta 0)
#remote: Use of 'master' branch is deprecated on git.bioconductor.org;
#remote:   Access to 'master' will eventually be revoked.
#remote:   Please use 'devel' to update Bioconductor devel version.
#remote:   See: http://contributions.bioconductor.org/branch-rename-faqs.html
#To git@git.bioconductor.org:packages/SurfR.git
#acf8a50..5a0f525  master -> master

git push upstream master:devel

#Total 0 (delta 0), reused 0 (delta 0)
#To git@git.bioconductor.org:packages/SurfR.git
#acf8a50..5a0f525  master -> devel


I'll move to devel as prompt here http://contributions.bioconductor.org/branch-rename-faqs.html.
Hope it solves the issue.

Best,
Aurora

**Comment by lshep** [comment]
Yes please only push to the devel branch on git.bioconductor.org.  We do encourage you to change your default branch on github to devel to be consistent and may also help clear things up.  I'll monitor and see when there are future pushes.  In the meantime, I'll kick off a manual build to hopefully get the current version's report.

**Comment by auroramaurizio** [comment]
Dear Pete,
Thank you for taking action towards the resolution of the "Rhdf5lib-mac" issue in the bc build report (https://github.com/Bioconductor/BBS/issues/245). The build/check report was successful on the Linux machine. 
Please let us know if there is something else we can do in the meanwhile.
Best, Aurora

**Comment by PeteHaitch** [comment]
Hi Aurora,

You might address some of the `NOTE`s raised by running `BiocCheck::BiocCheck()` on the package (the results are listed in the build report).
I'm not too worried about the code styling recommendations, that's largely personal preference, but there are other recommendations that are worth following.

I'll post my full review by the end of next week at the latest.

Cheers,
Pete

**Comment by auroramaurizio** [comment]
Dear Pete, 
we've started addressing the BiocCheck notes, but we've encountered a new issue. The code is now failing during both linux and mac bioconductor builds/checks. We tested it on our local computers and cluster and it works fine. The issue appears to be centered around the function "Gene2SProtein" in the vignette quickstart, which wasn't affected by the recent small code updates. Since this function downloads a table from an url (the url is still accessible and functioning) could the error be due to newly occurred connectivity issues ?

Thank you for your assistance,
Aurora

**Comment by PeteHaitch** [comment]
You might try triggering a new build of the package to see if it was an intermittent connectivity issue. 
You can do this by bumping the version number and pushing to the BioC git server.

**Comment by lshep** [comment]
FWIW -- to prevent this sort of thing from occurring too we highly recommend caching anything downloaded from external urls.

**Comment by PeteHaitch** [comment]
@auroramaurizio I'm going on leave today and will be offline until Oct 9.
Your package will be re-assigned by @lshep to someone else for review.

**Comment by auroramaurizio** [comment]
Dear @PeteHaitch,
Thank you for your time and attention to our package.

The check is now successful on the linux builder.
I see that the issue https://github.com/Bioconductor/BBS/issues/245 about the Iconway Mac builder configuration is still open. 
In the meanwhile, as you suggested me, I addressed several nebbiolo2 BiocCheck notes.

Best,
Aurora

**Comment by auroramaurizio** [comment]
Dear @lshep,

I am reaching out to follow up on the communication from our former reviewer, Peter Hickey. Peter informed us that he is currently on leave and that a new reviewer will be assigned to our package.

With the upcoming 3.18 Bioconductor package release approaching, do you think SurfR package could be considered for revision even if it passes the Linux nebbiolo2 check but 
Iconway macOS check is currently infeasible due to the builder issue outlined here https://github.com/Bioconductor/BBS/issues/245#issuecomment-1721605029?
We tested it on our macOS  laptops and the checks were successful (e.g. https://github.com/Bioconductor/Contributions/issues/3105#issuecomment-1713954422). 

I appreciate your input and guidance.

Thank you,
Aurora

**Comment by ococrook** [comment]
Hi @auroramaurizio 

Did you recieved and respond to a review from Peter? I cant see anything explicit above but dont want to make you do work you have already done? In the meantime. The package is welll put together and like the vignette

- [ ] It worth stating the R dependency
- [ ] Is the uncommented code needed? Usually best to delete this to avoid confusion
- [ ] Really worth having a comphrehensive .readme so that that someone can understand if they are interested in the package before loading it? What does tha package do exactly.

Thanks,
Olly

**Comment by auroramaurizio** [comment]
Hi @ococrook,
I haven't received a package review yet. 
Thank you for your advice. I will address the points you mentioned and get back to you!

Best,
Aurora

**Comment by auroramaurizio** [comment]
Dear @ococrook,
I bumped the new package version addressing your points, in particular, I:

- cleaned the code, removing commented lines
- added the README.md
- stated the R dependency in the README.md

Thank you, let me know how it looks now!
Aurora

**Comment by ococrook** [comment]
Thanks Aurura, trying to make may way through the package but I get stuck in the vignette

```{r,}
> # Download metadata from GEO
> mGSE121810 <- GEOmetadata(GSE = "GSE121810")
```
Warning: cannot open file 'GSE121810_series_matrix.txt.gz': No such file or directoryError in file(file, "rt") : cannot open the connection

Some comments on the R code:

- [ ]  You have some contructions like `5:N`, we generally recommend using something like `seq_along`, `seq_int ` etc.
- [ ] Still examples of code with comments that are likely to cause confusion: e.g. `grouped  <-
    exploded %>%
    #dplyr::group_by(GeneID) %>%
    #dplyr::summarise(temp = toString(term)) %>%
    #dplyr::ungroup()
    group_by(GeneID) %>%
    summarise(temp = toString(term)) %>%
    ungroup()`
- [ ] Is this sequence going backwards? `as.factor(dim(pathways.dataframe)[1]:1)` Im a bit confused as to what this coded is intneded to do.
- [ ] Try to be consistent with white spaces, espcially after columns or operations
- [ ] I'm not sure whats happening here, did you redefine `log` and then use the base `log` again? `log <- sort(list.files("log/", pattern = "*.xlsx"))

  # ---- surfy database versioning ----

  if (!Surfy_version %in% c("log", "new")) {
    stop("The specified database version is not available. \n  Choose between log or new")
  }

  if (length(log)>0 & Surfy_version == "log") {
    ST <- read.xlsx(xlsxFile = paste0(".log/", log[length(log)]),
                    startRow = 1)
  }`

- [ ] I would stick with `<-` for assignment note sure why you switch to using `=` for some functions? 
- [ ] I usually don't advise to use `as.factor` and suggest to convert to a factor more explicity with `factor` specificying the levels you expect
- [ ] You have lots of imports in the description are these necessary?

I think the R code in general could be cleaned up but I'm very happy with the documentation and vignette (once I can getting working)

**Comment by auroramaurizio** [comment]
Thank you @ococrook for your careful code revision.

- I'm sorry to hear that you are having problems in the vignette with the GEOmetadata function. Most probably you are facing the Rhdf5lib Mac Iconway builder issue reported here https://github.com/Bioconductor/BBS/issues/245#issuecomment-1721605029. hpages clearly defines the problem and the steps to solve it. This issue is also the reason why SurfR fails the Mac check on the Iconway buidler but i.e. runs successfully on our Mac laptops https://github.com/Bioconductor/Contributions/issues/3105#issuecomment-1713954422. Hope you'll manage to fix it!

- All the functions should be cleared from code commented lines now.

- I removed the line "as.factor(dim(pathways.dataframe)[1]:1)" from the "Enrichment_barplot" function. It wasn't used.

- I spotted some instances like  1:... where seq_len(),  seq_along(), etc. would fit better and used these constructs.

- In the Gene2SProtein function we allow the user to select the version of the Surfy data frame he wishes to use. In other words, to decide if he wants to use an xlsx table already downloaded from Surfy's URL or to download it again, choosing between "log" and "newest" Surfy_version. The line "log <- sort(list.files("log/", pattern = "*.xlsx"))" screens the ".log/" directory and lists in "log" all the xlsx files present in the directory in the variable "log". To avoid confusion I renamed the variable log as log_list. 

- I improved code styling: commas now always have a space after, spaces are present around infix operators, I always use double quotes, etc.

- I stuck with <- for assignments

- I replaced as.factor with factor.

- I agree there are quite a few imports. The package aims to allow the user to perform an end-to-end protein annotation analysis from data retrieval to DGE, meta-analysis, annotation, plotting, etc., and each of these steps requires  dedicated packages

Aurora

**Comment by lshep** [comment]
**R CMD check**
- [ ] @auroramaurizio  It looks like there were still ERROR in the linux build report unrelated to the one mentioned above on MAC.  It may have just been temporary but when I run it manually on the builder
```
> dbs
[1] "GO_Biological_Process_2021" "WikiPathways_2016"         
[3] "MSigDB_Hallmark_2020"      
>  up_genes <- c("RUNX1", "DLK1", "TOP2A", "EPCAM", "GATA1", "KDR")
    dfList[["fdr_up"]] <- enrichR::enrichr(up_genes, dbs)
   
Must specify at least one of url or handle
>   dw_genes <- c("CD275", "COL1A1", "COL1A2","LUM", "SOX9")
    dfList[["fdr_down"]] <- enrichR::enrichr(dw_genes, dbs)
  
Must specify at least one of url or handle
>    Enrichment_barplot(dfList,
                       enrich.databases = dbs,
                       p_adj = 0.01, num_term = 5, cond = "UP")
Error in Enrichment_barplot(dfList, enrich.databases = dbs, p_adj = 0.01,  : 
  Selected Enrich databases not present in your enrichment analysis.
In addition: Warning message:
In Enrichment_barplot(dfList, enrich.databases = dbs, p_adj = 0.01,  :
  GO_Biological_Process_2021WikiPathways_2016MSigDB_Hallmark_2020not present in your Enrichment analysis.
>     Enrichment_barplot(dfList,
                       enrich.databases = dbs,
                       p_adj = 0.01, num_term = 5, cond = "DOWN")
Error in Enrichment_barplot(dfList, enrich.databases = dbs, p_adj = 0.01,  : 
  Selected Enrich databases not present in your enrichment analysis.
In addition: Warning message:
In Enrichment_barplot(dfList, enrich.databases = dbs, p_adj = 0.01,  :
  GO_Biological_Process_2021WikiPathways_2016MSigDB_Hallmark_2020not present in your Enrichment analysis.

```

**BiocCheck**
- [ ] The R version WARNING should also be corrected;  we can not verify that this will work on such an outdated version of R.  

- [ ] avoiding `cat` and use `message` instead so users that don't want verbose output can suppressMessages. Found in 

```
cat() in R/Enrichment.R (line 66, column 5)
cat() in R/Enrichment.R (line 74, column 5)
cat() in R/Enrichment.R (line 91, column 5)
```

- [ ] Please explain why the suppressWarnings are justified 
```
suppressWarnings() in R/Annotate_SPID.R (line 156, column 3)
suppressWarnings() in R/SVenn.R (line  41, column 3)

```

**Comment by auroramaurizio** [comment]
Dear @lshep,
Thank you for your attention to SurfR package. I'm sorry to hear you got an ERROR on your manual linux build.  I can confidently say the ERROR you are facing now is due to a temporary connection problem with the EnrichR database. If it happens I re-run the command after a few minutes (1, 2 minutes) and the connection with EnrichR is re-established. I now added a line in the vignette about this potential temporary issue.

The automatic linux builds and checks on nebbiolo2 of SurfR_0.99.20.tar.gz, but also SurfR_0.99.19.tar.gz, SurfR_0.99.18.tar.gz etc. were all successful. 
I can assure we also tested it on our linux cluster and 4 different MacOS laptops and it worked well without any errors. If It may be helpful I can send the session info of each. 

We tested the package on R versions 4.1 and greater. I now explicitly report all tested versions (4.1, 4.2, and 4.3) in the README. I do not get an Rversion warning, unfortunately, I can not reproduce it.

I replaced cat with message in R/Enrichment.R (line 66, line 74, line 91). 

I removed suppressWarnings() in SVenn function, while I kept it in Annotate_SPID. I added a comment above suppressWarnings() in Annotate_SPID to provide context to the user and justify its usage.

Thank you,
Aurora

**Comment by lshep** [comment]
Unfortunately it seems to be consistently ERRORing.  Is there any way to make the package more dependable or able to fail out gracefully if the connection cannot be established. 

Because there are data objects R during the build stage will automatically add an R version because of incompatiblity with pervious version saved objects. To fix this ERROR please put the following in the DESCRIPTION  `R (>= 4.3.0)` in the Depends field.  

Thank you for the other changes.

**Comment by auroramaurizio** [comment]
Dear @lshep,
thank you for your advice. 
I added the Depends field in DESCRIPTION, fixed the issue with the failing function and now the check on the nebbiolo2 linux builder is successful.

Aurora

**Comment by ococrook** [comment]
Sorry for the delay here, could you add the bioconductor install instruction. Could you also trigger a new build? I would like to see whats happening there more recently.

**Comment by auroramaurizio** [comment]
Hi @ococrook. I triggered a new build adding the bioconductor install instruction. The nebbiolo1 linux build is successful. Therefore I think the ERROR message highlighted on top of this page is still the same old one (rhdf5lib problem on bioconductor Mac builder)  https://github.com/Bioconductor/BBS/issues/245 https://github.com/Bioconductor/BBS/issues/245#issuecomment-1832476492.

**Comment by jwokaty** [comment]
Sorry if this has been previously been mentioned in your review, but I noticed while trying to manually run `R CMD build SurfR` on the 3.19 intel mac for https://github.com/Bioconductor/BBS/issues/245 that it failed because it tries to `wget` a resource; however, `wget` isn't available by default. It would be better to use an R function if you have to retrieve a resource and use caching. See https://contributions.bioconductor.org/data.html?q=download#other-data for additional guidance.

**Comment by auroramaurizio** [comment]
Thank you @jwokaty. I followed your advice about wget. Now the ERROR message has been replaced by a WARNING. I think the WARNING is still from the MAC builder but I see only the Linux report where no warnings are present.

**Comment by lshep** [comment]
sorry  I will kick off a manual build; the mac builder has been added back into the report and you should see it shortly.

**Comment by auroramaurizio** [comment]
Hi @ococrook, I'm happy to tell you that the report is now successful on both Mac and Linux builders!

**Comment by ococrook** [comment]
Hi @auroramaurizio ,

Sorry this has been a slow proces, a final few things before we can finish this off:

Can you make the version of the package match between the NEWS file and Description?

Can you switch to the Biocstyle for the vignette?

youll need to add the following to the header
```{}
output:
  BiocStyle::html_document:
    toc_float: true
```

**Comment by auroramaurizio** [comment]
Hi @ococrook, 
I updated the vignette and the NEWS file as per your suggestions. 
Thank you! Aurora

**Comment by auroramaurizio** [comment]
Dear @ococrook, 
I hope you're doing well. 
Do you think there is a chance SurfR package can be included in the upcoming April 2024 Bioconductor 3.19 release? Please let me know if I should do anything to proceed. 
Thank you, 
Best, 
Aurora

**Comment by ococrook** [comment]
Hi Aurora,

Not sure what happen but I recall trying to accept this package several times now. Hopefully, it gone through now.

**Comment by auroramaurizio** [comment]
Thank you very much Oliver!
I wish you a great day,
Aurora

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/auroramaurizio.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("SurfR")`. The package 'landing page' will be created at

https://bioconductor.org/packages/SurfR

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
